File: src/com/woezelmann/showas/ShowAs.java
package com.woezelmann.showas;

import com.intellij.lang.Language;
import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.project.Project;
import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiFile;
import com.intellij.psi.PsiFileFactory;
import com.intellij.psi.codeStyle.CodeStyleManager;

public abstract class ShowAs extends AnAction {

    private final Language language;

    public ShowAs(Language language) {
        this.language = language;
    }

    @Override
    public void actionPerformed(AnActionEvent event) {

        String text = fetchValue(event);

        if (text == null) {
            return;
        }

        final Project project = event.getProject();
        final PsiFile psiFile = PsiFileFactory.getInstance(project).createFileFromText(language, preProcessValue(text));
        final PsiElement reformat = CodeStyleManager.getInstance(project).reformat(psiFile.copy());

        EditorWrapper editorWrapper = new EditorWrapper(language, project, reformat.getText());
        editorWrapper.show();
    }

    public abstract String fetchValue(AnActionEvent event);

    public String preProcessValue(String value) {
        return value;
    }
}

File: src/main/kotlin/com/woezelmann/showas/ShowAs.kt
package com.woezelmann.showas

import com.intellij.json.JsonLanguage
import com.intellij.lang.Language
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.project.Project
import com.intellij.psi.PsiFile
import com.intellij.psi.PsiFileFactory
import com.intellij.psi.codeStyle.CodeStyleManager


abstract class ShowAs(val language: Language) : AnAction() {

    override fun actionPerformed(event: AnActionEvent) {

        val text: String? = fetchValue(event)

        if (text == null) {
            return
        }

        val project: Project? = event.getProject()
        val psiFile: PsiFile = PsiFileFactory.getInstance(project).createFileFromText(language, text)

        val reformat = CodeStyleManager.getInstance(project!!).reformat(psiFile.copy())

        val editorWrapper = EditorWrapper(JsonLanguage.INSTANCE, project, reformat.text)
        editorWrapper.show()

    }

    abstract fun fetchValue(event: AnActionEvent?): String?

}